include/save.h

```diff
@@ -401,13 +401,55 @@ typedef enum SceneLayer {
 } SceneLayer;
 
 #define IS_CUTSCENE_LAYER (gSaveContext.sceneLayer >= SCENE_LAYER_CUTSCENE_FIRST)
+#define GET_CUTSCENE_LAYER(index) (SCENE_LAYER_CUTSCENE_FIRST + (index & 0xF))
 
 typedef enum LinkAge {
     /* 0 */ LINK_AGE_ADULT,
     /* 1 */ LINK_AGE_CHILD
 } LinkAge;
 
 
+// Values that indicate that no cutscene script should be played
+// (or that an "unscripted" cutscene is playing). The names "night"
+// and "day" are leftover from the original meaning of the cutscene
+// index (called "day_time"), and only map select uses these values
+// to indicate time of day (to set gSaveContext.save.nightFlag).
+#define CS_INDEX_NONE 0x0000
+#define CS_INDEX_NIGHT CS_INDEX_NONE
+#define CS_INDEX_DAY 0x8000
```

Comment:
I would prefer these to be defined separately with their own explanations, and no aliasing to occur:
```suggestion
// Usage in Map Select suggests that `gSaveContext.save.cutsceneIndex` was,
// at one point in development, a variable related to the time. 
// This is further supported by debug strings showing that the original variable name was "day_time".
// These defines exist for the rare cases in the codebase where cutsceneIndex is treated as a time value.
// In practice, both values do not lead to a cutscene being played.
// See below for the more common usage of value zero.
#define CS_INDEX_NIGHT 0x0000
#define CS_INDEX_DAY 0x8000

// Indicates that no scene layer cutscene is playing (or should be played).
// While this value overlaps with `CS_INDEX_NIGHT` defined above, it is not related to the time.
// This is the more common interpretation of this value, as much of the codebase uses zero to mean "no cutscene"
// except for Map Select.
#define CS_INDEX_NONE 0x0000
```